forwards_paper.tex

```diff
@@ -240,6 +241,46 @@ \subsection*{Simulation benchmarks}
 
 \plr{Insert figures/tables of performance here and describe them.}
 
+The total run times for simulations with selection are shown in Figure~\ref{fig:runtimes_selection}.  When tracking
+neutral mutations (instead of the ARG), run times increase dramatically with increasing region size ($4Nr = 4Nu$ for
+these simulations).  With \fwdpy{}, simulations with $N=5 \times 10^4$ timed out for region sizes larger than $10^3$.
+With \simupop{}, we were only able to run simulations with $N=10^3$ and region sizes up to $10^4$.  ARG tracking
+dramatically improves run times for both \fwdpy{} and \simupop{} (right column of Figure~\ref{fig:runtimes_selection}).  For parameters that ran to completion when tracking neutral mutations and when tracking
+the ARG, the relative speedup due to ARG tracking is up to $\approx 50$ fold (Figure~\ref{fig:relative_speedup_selection}).  For \fwdpy{}, ARG tracking reduces runtimes more for larger population sizes
+(Figure~\ref{fig:relative_speedup_selection}).  We could not make such an observation with \simupop{}, which tended to
+run out of memory on our system.  Note the log scale of the x axis in Figure~\ref{fig:runtimes_selection}, which
+partially obscures an important observation that the run times appear approximately linear in region size when ARG
+tracking.  The reason for this behavior is that the simulations are generating a constant number of new edges on average
+each generation, and the number of edges is a function of $\rho$ in expectation.
+
+We also performed a more limited set of simulations without natural selection.  The total run times are shown in
+Figure~\ref{sfig:rawspeed_nosel} and show the same qualitative behavior as simulations with selection
+(Figure~\ref{fig:runtimes_selection}).  The relative improvement due to ARG tracking is again substantial in these
+simulations (Figure~\ref{sfig:speedup_nosel}).  In fact, we see more of a benefit to ARG tracking here with \fwdpy11{}
+than we did with selection (Figure~\ref{fig:relative_speedup_selection}).  The reason is that the fitness function exits
+close to instantly in simulations without selection that are based on \fwdpp{}, meaning that \fwdpy{} is doing little
+more than generating random numbers and book-keeping in Figures~\ref{sfig:rawspeed_nosel}
+and~\ref{sfig:speedup_nosel}.
```

Comment:
we should make clear elsewhere (the discussion?) why applying this to neutral simulations is useful.

Reply:
You mean non-WF kinds of models?

Reply:
yep.

huh, looking at the plots, it looks to me like the relative speedup is about the same for fwdpy11 with neutral vs selected simulations.

Reply:
![screenshot from 2017-12-22 11-07-21](https://user-images.githubusercontent.com/1046249/34309780-afd406f6-e708-11e7-95ed-43ca7adc1d8b.png)

Yes -the speedup is the same with and without selected mutations.  I don't get it?

Reply:
I take that back - not the same, but not as different as I thought they would be:
![screenshot from 2017-12-22 11-12-44](https://user-images.githubusercontent.com/1046249/34309856-1a1af4c0-e709-11e7-98ee-3df40a94d8bc.png)

The neutral ones have a speedup of like 10%-50% larger.

Reply:
Yep--that's right.  For the most common use of forward sims, this illustrates why it is problematic to show benchmarks based only on neutral sims.  The run times are longer with selection.  There are two differences between the two types of sim (in fwdpp, at least).  First, more mutations in the sims with selection--thousands for our params. Second, the fitness function  call is really close to a no-op for neutral sims, but takes a big chunk of run time with selection.

Reply:
I get it that the *absolute* time should be very different for with and without selection, but I thought that the speedup, i.e., the relative no-ARG/ARG time should be much larger without selection?